app/jobs/scrape_catalogues_job.rb

```diff
@@ -1,18 +1,20 @@
+require 'capybara'
+require 'capybara/poltergeist'
+
 class ScrapeCataloguesJob < ActiveJob::Base
+  include Capybara::DSL
   queue_as :default
 
-  require 'capybara'
-  require 'capybara/poltergeist'
-
   def perform(catalogue_to_add)
     num_retry = 0
+    max_retry = 4
     session_wait = 7
 
     catalogue_to_add.each do |catalogue_num|
       begin
 
         #visit page
-        visit catalogue_url(catalogue_num, nil)
+        visit catalogue_url(catalogue_num)
         
         #creating new catalogue

```

Comment:
No need for comment here, code line immediately below succinctly explains what's happening.